integration_test/third_party_apps_test.go

```diff
@@ -172,65 +172,61 @@ func installAgent(ctx context.Context, logger *logging.DirectoryLogger, vm *gce.
 	return nonRetryable, agents.InstallPackageFromGCS(ctx, logger, vm, agents.OpsAgentType, packagesInGCS)
 }
 
-// expectedLogs encodes a series of assertions about what data we expect
-// to see in the logging backend.
-type expectedLogs struct {
-	// Note on tags: the "yaml" tag specifies the name of this field in the
-	// .yaml file.
-	LogEntries []expectedLog `yaml:"log_entries"`
+type logFields struct {
+	Name        string `yaml:"name" validate:"required"`
+	ValueRegex  string `yaml:"value_regex"`
+	Type        string `yaml:"type" validate:"required"`
+	Description string `yaml:"description" validate:"required"`
 }
+
 type expectedLog struct {
-	LogName string `yaml:"log_name"`
-	// Map of field name to a regex that is expected to match the field value.
-	// For example, {"jsonPayload.message": ".*access denied.*"}.
-	FieldMatchers map[string]string `yaml:"field_matchers"`
+	LogName string      `yaml:"log_name" validate:"required"`
+	Fields  []logFields `yaml:"fields" validate:"required"`
+}
+
+type integrationMetadata struct {
+	ExpectedLogs    []expectedLog           `yaml:"expected_logs"`
+	ExpectedMetrics []common.ExpectedMetric `yaml:"expected_metrics"`
 }
 
-// constructQuery converts the given map of:
+// constructQuery converts the given struct of:
 //   field name => field value regex
 // into a query filter to pass to the logging API.
-func constructQuery(fieldMatchers map[string]string) string {
+func constructQuery(fields []logFields) string {
 	var parts []string
-	for field, matcher := range fieldMatchers {
-		parts = append(parts, fmt.Sprintf("%s=~%q", field, matcher))
+	for _, field := range fields {
+		if field.ValueRegex != "" {
+			parts = append(parts, fmt.Sprintf("%s=~%q", field.Name, field.ValueRegex))
```

Comment:
I noticed that some of the `value_regex` contain spaces so the regex needs to be quoted:

```suggestion
			parts = append(parts, fmt.Sprintf(`%s=~"%q"`, field.Name, field.ValueRegex))
```